File: src/main/java/toby/jpa/persistence/IConfigPersistence.java
package toby.jpa.persistence;

import toby.jpa.dto.ConfigDto;

import java.util.List;

public interface IConfigPersistence {
    public List<ConfigDto> listConfig();

    public ConfigDto getConfigByName(String name, String guildId);

    public ConfigDto createNewConfig(ConfigDto configDto);

    public ConfigDto updateConfig(ConfigDto configDto);
}

File: src/main/java/toby/jpa/persistence/IConfigPersistence.kt
package toby.jpa.persistence;

import toby.jpa.dto.ConfigDto;

import java.util.List;

public interface IConfigPersistence {
    List<ConfigDto> listAllConfig();
    List<ConfigDto> listGuildConfig(String guildId);
    ConfigDto getConfigByName(String name, String guildId);
    ConfigDto createNewConfig(ConfigDto configDto);
    ConfigDto updateConfig(ConfigDto configDto);
    void deleteAll(String guildId);
    void deleteConfig(String guildId, String name);
}
